l1-contracts/test/Decoder.t.sol

```diff
@@ -102,6 +121,15 @@ contract DecoderTest is Test {
       "Invalid public input hash"
     );
 
+    for (uint256 i = 0; i < l2ToL1Msgs.length; i++) {
+      assertEq(
+        l2ToL1Msgs[i], bytes32(uint256(0x300 + 32 * (1 + i / 2) + i % 2)), "Invalid l2ToL1Msgs"
```

Comment:
bruh we gotta automate test generation, can you lift the value calc up a line so its a bit more readable

Reply:
Will be adding the test in the integration test for L1 publisher when adding in rollup, then tested in there for the automatic. 
For our use, even if I like foundry more, TS seems like a better fit to have the tests because we will have so much encoding and stuff to deal with 😬 I only really see our foundry tests right now as useful for better debugging when building and for checking stuff without all the encoding/decoding.